src/utils/WorkspacesUtils.js

```diff
@@ -55,6 +67,69 @@ const patchWorkspaceWithDatasetManagerConfiguration = (workspace) => {
   }
 };
 
+const checkDatasetManagerConfiguration = (workspace) => {
+  const logWarning = (warning) => console.warn(`Dataset manager configuration: ${warning}`);
+  const config = workspace?.webApp?.options?.datasetManager;
+  if (config == null) return;
+  if (!(config instanceof Object)) {
+    logWarning('"datasetManager" must be an Object');
+    return;
+  }
+
+  const { graphIndicators, categories, queries } = config;
+
+  if (queries != null && !(queries instanceof Array)) {
+    logWarning('property "queries" must be an Array');
+  } else {
+    const allQueriesIds = [];
+    queries.forEach((query, index) => {
+      if (query.id == null) logWarning(`in queries, item #${index} has no id`);
+      else if (allQueriesIds.includes(query.id))
+        logWarning(`in queries, item #${index} uses a query id that already exists (id: ${query.id}).`);
+      else allQueriesIds.push(query.id);
+
+      if (query.query == null) logWarning(`in queries, item #${index} (id: ${query?.id}) has no query statement`);
+    });
+  }
+
+  const allKpisIds = [];
+  const isQueriesValid = queries != null && queries instanceof Array;
```

Comment:
I think we can check only `queries instanceof Array` because null can't be one :slightly_smiling_face:

Reply:
You're right, thanks :slightly_smiling_face: